capstone/src/main/webapp/data_viz.js

```diff
@@ -168,4 +153,132 @@ function mouseInToRegion(e) {
 function mouseOutOfRegion(e) {
   // Reset the hover state, returning the border to normal
   e.feature.setProperty('state', 'normal');
+}
+
+/* get County data from NYT in the form of a dictionary to put on the graph*/
+//date,county,state,fips,cases,deaths,confirmed_cases,confirmed_deaths,probable_cases,probable_deaths
+function getCounties(variable){
+  fetch('/getCountyData').then(response => response.text()).then((output) => {
+    let nyt_data = {};
+    let nyt_data_by_state = {};
+    let state_names = [];
+
+    let data_lines = JSON.parse(output);
+    for(let i = 0; i < data_lines.length; i++){
+      if(data_lines[i] !== null){
+        let parsed_data = data_lines[i].split(',');
+        if(parsed_data.length === 10){
+          let date = parsed_data[0];
+          let county = parsed_data[1];
+          let state = parsed_data[2];
+          let fips = parsed_data[3];
+          let cases = parsed_data[4];
+          let deaths = parsed_data[5];
+          let confirmed_cases = parsed_data[6];
+          let confirmed_deaths = parsed_data[7];
+          let probable_cases = parsed_data[8];
+          let probable_deaths = parsed_data[9];
+          let key = county + '-' + fips;
+          //note that there may be blank '' attributes
+          let county_data = {"date": date, "state": state, "fips": fips, "cases": cases,
+            "deaths": deaths, "confirmed_cases": confirmed_cases, "confirmed_deaths": confirmed_deaths, 
+            "probable_cases": probable_cases, "probable_deaths": probable_deaths};
+          nyt_data[key] = county_data;
+          county_data["county"] = county;
+          county_data["fips"] = fips;
+          if(nyt_data_by_state[state] !== undefined){
+            nyt_data_by_state[state].push(county_data);  
+          }else{
+            state_names.push(state);
+            nyt_data_by_state[state] = [county_data];
+          }
+        }
+      }
+    }
+
+    let state_counts = {};
+    if(variable === "cases"){
+      state_counts = setStateWideCases(state_names, nyt_data_by_state);
+    }
+    if(variable === "deaths"){
+      state_counts = setStateWideDeaths(state_names, nyt_data_by_state);
+    }
+    setData(state_counts);
+    let range = findMinMax(state_names, state_counts);
+    localStorage.setItem("min", range.min);
+    localStorage.setItem("max", range.max);
+    document.getElementById('data-min').textContent = range.min.toLocaleString();
+    document.getElementById('data-max').textContent = range.max.toLocaleString();
+  });
+}
+
+/* finds the min and max case or death counts for the legend */
+function findMinMax(state_names, state_counts){
+  let min = Number.MAX_VALUE;
+  let max = -Number.MAX_VALUE;
+  for(let i = 0; i < state_names.length; i++){
+    if(STATES_DISPLAYED.indexOf(state_names[i]) > -1){
+      let count = state_counts[state_names[i]];
+      if(min > count){
+        min = count;
+      }
+      if(max < count){
+        max = count;
+      }
+    }
+  }
```

Comment:
Maybe this whole function could be replaced with Math.max and Math.min functions. Usually library functions are faster (e.g. running assembly under the covers).